File: examples/server/routes/languages/kotlin/yon.kt
class Handler {
    companion object {
        fun GET(request: Map<String, Any?>): Any? {
            return when (statusCode(request)) {
                "411" -> mapOf("code" to "411", "detail" to "length required")
                "412" -> mapOf("code" to "412", "detail" to "precondition failed")
                "413" -> mapOf("code" to "413", "detail" to "payload too large")
                "414" -> mapOf("code" to "414", "detail" to "uri too long")
                "415" -> mapOf("code" to "415", "detail" to "unsupported media type")
                else -> KotlinLanguageService().describe(request)
            }
        }

        @Suppress("UNCHECKED_CAST")
        private fun statusCode(request: Map<String, Any?>): String {
            val query = request["query"] as? Map<String, Any?> ?: return ""
            return when (val code = query["code"]) {
                is String -> code
                is Number -> code.toLong().toString()
                else -> ""
            }
        }
    }
}

File: examples/server/routes/languages/csharp/yon.cs
using System.Collections.Generic;
using System.Text.Json;

public class Handler
{
    private static readonly CSharpLanguageService Service = new();

    private static readonly Dictionary<string, Dictionary<string, object?>> Responses = new()
    {
        ["423"] = Response("423", "locked"),
        ["424"] = Response("424", "failed dependency"),
        ["425"] = Response("425", "too early"),
        ["426"] = Response("426", "upgrade required"),
        ["428"] = Response("428", "precondition required"),
    };

    public static Dictionary<string, object?> GET(JsonElement request)
    {
        var code = StatusCode(request);
        if (Responses.TryGetValue(code, out var response))
            return response;

        return Service.Describe(request);
    }

    private static string StatusCode(JsonElement request)
    {
        if (!request.TryGetProperty("query", out var query)
            || !query.TryGetProperty("code", out var raw))
            return "";

        return raw.ValueKind == JsonValueKind.Number
            ? raw.GetInt32().ToString()
            : raw.GetString() ?? "";
    }

    private static Dictionary<string, object?> Response(string code, string detail)
    {
        return new Dictionary<string, object?>
        {
            ["code"] = code,
            ["detail"] = detail,
        };
    }
}
